crates/rooch/src/commands/account/commands/update.rs

```diff
@@ -0,0 +1,140 @@
+    async fn execute(self) -> RoochResult<ExecuteTransactionResponseView> {
+        println!("{:?}", self.mnemonic_phrase);
+
+        let mut context = self.context_options.build().await?;
+
+        match BuiltinScheme::from_flag_byte(&self.crypto_schemes.flag()) {
+            Ok(scheme) => {
+                let existing_address =
+                    RoochAddress::from_str(self.address.as_str()).map_err(|e| {
+                        RoochError::CommandArgumentError(format!(
+                            "Invalid Rooch address String: {}",
+                            e
+                        ))
+                    })?;
+
+                let public_key: PublicKey = context
+                    .config
+                    .keystore
+                    .update_address_with_key_pair_from_scheme(
+                        &existing_address,
+                        self.mnemonic_phrase,
+                        scheme,
+                        None,
+                    )
+                    .map_err(|e| RoochError::UpdateAccountError(e.to_string()))?;
+
+                println!(
+                    "{}",
+                    AccountAddress::from(existing_address).to_hex_literal()
+                );
+                println!(
+                    "Generated a new keypair for an existing address on scheme {:?} [{existing_address}]",
+                    scheme.to_owned()
+                );
+
+                let validator_struct_arg: Box<StructTag> = match scheme {
+                    BuiltinScheme::Ed25519 => Ok(Box::new(StructTag {
+                        address: Ed25519Validator::MODULE_ADDRESS,
+                        module: Identifier::new(Ed25519Validator::MODULE_NAME.to_string()).unwrap(),
+                        name: Identifier::new(String::from(stringify!(Ed25519Validator))).unwrap(),
+                        type_params: vec![],
+                    })),
+                    BuiltinScheme::Ecdsa => Ok(Box::new(StructTag {
+                        address: EcdsaK1Validator::MODULE_ADDRESS,
+                        module: Identifier::new(EcdsaK1Validator::MODULE_NAME.to_string()).unwrap(),
+                        name: Identifier::new(String::from(stringify!(EcdsaK1Validator))).unwrap(),
+                        type_params: vec![],
+                    })),
+                    BuiltinScheme::Schnorr => Ok(Box::new(StructTag {
+                        address: SchnorrValidator::MODULE_ADDRESS,
+                        module: Identifier::new(SchnorrValidator::MODULE_NAME.to_string()).unwrap(),
+                        name: Identifier::new(String::from(stringify!(SchnorrValidator))).unwrap(),
+                        type_params: vec![],
+                    })),
+                    _ => Err(RoochError::CommandArgumentError(
+                        "Validator for this scheme is not implemented".to_owned(),
+                    )),
+                }?;
```

Comment:
This code can be a function of BuiltinScheme

Reply:
Will refactor it.